homeassistant/components/motionblinds_ble/cover.py

```diff
@@ -0,0 +1,717 @@
+
+    def async_register_speed_callback(
+        self, _speed_callback: Callable[[MotionSpeedLevel | None], None]
+    ) -> None:
+        """Register the callback used to update the speed level."""
+        self._speed_callback = _speed_callback
+
+    def async_register_connection_callback(
+        self, _connection_callback: Callable[[MotionConnectionType | None], None]
+    ) -> None:
+        """Register the callback used to update the connection."""
+        self._connection_callback = _connection_callback
+
+    def async_register_signal_strength_callback(
+        self, _signal_strength_callback: Callable[[int | None], None]
+    ) -> None:
+        """Register the callback used to update the signal strength."""
+        self._signal_strength_callback = _signal_strength_callback
+
+    @property
+    def extra_state_attributes(self) -> Mapping[str, str]:
+        """Return the state attributes."""
+        return {ATTR_CONNECTION_TYPE: self._attr_connection_type}
+
+    async def before_command_function(self, *args, **kwargs) -> None:
+        """Run some code before executing any command."""
+        if self._attr_connection_type is MotionConnectionType.CONNECTED:
+            self.async_refresh_disconnect_timer()
+
+    # Decorator
+    async def run_command_function(
+        self,
+        func: Callable,
+        ignore_end_positions_not_set: bool = False,
+        *args,
+        **kwargs,
+    ) -> bool:
+        """Run some code before executing any command that moves the position of the blind."""
+        await self.before_command_function(*args, **kwargs)
+        if self._attr_connection_type is not MotionConnectionType.CONNECTED:
+            self._use_status_position_update_ui = False
+        return await func(
+            self,
+            *args,
+            ignore_end_positions_not_set=ignore_end_positions_not_set,
+            **kwargs,
+        )
+
+    # Decorator
+    async def no_run_command_function(self, func: Callable, *args, **kwargs) -> bool:
+        """Run some code before executing any command that does not move the position of the blind."""
+        await self.before_command_function(*args, **kwargs)
+        if self._attr_connection_type is not MotionConnectionType.CONNECTED:
+            self._use_status_position_update_ui = True
+            self.async_update_running(MotionRunningType.STILL)
+        return await func(self, *args, **kwargs)
+
+
+class PositionBlind(GenericBlind):
+    """Representation of a blind with position capability."""
+
+    _attr_supported_features: CoverEntityFeature | None = (
+        CoverEntityFeature.OPEN
+        | CoverEntityFeature.CLOSE
+        | CoverEntityFeature.STOP
+        | CoverEntityFeature.SET_POSITION
+    )
+
+    @run_command
+    async def async_open_cover(
+        self, ignore_end_positions_not_set: bool = False, **kwargs: Any
+    ) -> None:
+        """Open the blind."""
+        _LOGGER.info("(%s) Opening", self.config_entry.data[CONF_MAC_CODE])
```

Comment:
I don't think this should be a `info` level log message. The same goes for any other messages that are logged each time default behaviour is executed.